services/web/server/src/simcore_service_webserver/projects/projects_utils.py

```diff
@@ -286,3 +294,69 @@ def any_node_inputs_changed(
                     )
                     return True
     return False
+
+
+def get_frontend_node_outputs_changes(
+    new_node: dict[str, Any], old_node: dict[str, Any], filter_keys: set[ServiceKey]
+) -> FrontendOutputsChanges:
+    """if node is a specific type it checks if outputs changed"""
+    nodes_keys = {old_node.get("key"), new_node.get("key")}
+    if not (len(nodes_keys) == 1 and nodes_keys.pop() in filter_keys):
+        return FrontendOutputsChanges(changed=False, keys=set())
+
+    log.debug("Comparing nodes %s %s", new_node, old_node)
+    outputs_changed = new_node.get("outputs") != old_node.get("outputs")
+
+    def _get_outputs_keys(node_data: dict[str, Any]) -> set[str]:
+        outputs = node_data.get("outputs", {})
+        if outputs is None:
+            return set()
+        return set(outputs.keys())
+
+    changed_keys = _get_outputs_keys(new_node).symmetric_difference(
```

Comment:
So by "changes" here you refer to which output key-names not in common between new and the old node. Regardless of its content (e.g. one output might also change its value), right?

Reply:
Yes, you are correct. That's what I want to detect. There was an issue with this approach and I had to scrap it.

The new method correctly computes what I want. The symmetric_difference only took into account the keys of the dict, not the values. I could not use it on the values.